### DISCUSSION TITLE
Adding support for H265 and JPEG

### DISCUSSION THREAD

**Discussion by helu-milestone** [sipsorcery-org/sipsorcery#1254] [General]
Hello SIP Sorcery

Are there any work going to to add support for H265 and JPEG?

We are considering to look into it and would just know if this would clash with someone else who are doing the same or a strategic decision to not include these codecs?

**Comment by sipsorcery** [comment]
As far as RTP packetisation goes:

 - JPEG: There is a [rudimentary implementation](https://github.com/sipsorcery-org/sipsorcery/blob/2ab8d55997464e558b85ceb196314bc0577e5d08/src/net/RTP/Packetisation/RtpVideoFramer.cs#L132).
 - H265: Nothing implemented or in progress fro RFC7798.
 
 As far as the codecs go both are almost certainly available via FFmpeg and could theoretically be wired up in the [SIPSorcery.FFmpeg](https://github.com/sipsorcery-org/SIPSorceryMedia.FFmpeg) repo.

**Reply by helu-milestone** [reply]
I tried to setup the basic implementation, but in the other end there is no depacktization available when receiving the JPEG stream. 

We have talked about FFmpeg before, but we need to bundle it with our software solution and our legal department are not happy about the LGPL license. But it might be worth it in this case.

**Reply by sipsorcery** [reply]
Yes, JPEG streams are very inefficient compared to modern video streaming codecs such as H265 and VP9 so not very popular. The only reason there's any packetisation implementation in this library is from an RTSP implementation 10-15 years ago.

If you don't want to use FFmpeg you could use libvpx, which is BSD licensed. There is even a very crude, and very poorly performing attempt, at a dotnet port here https://github.com/sipsorcery-org/VP8.Net. H264 and H265 require commercial licenses although they are often covered by the OS manufacturer.

**Reply by Terricide** [reply]
Cisco covers licensing if openh264.dll is downloaded from them.

https://www.openh264.org/

There are c# wrappers for it.
